### ISSUE TITLE
Twig is required unnecessary

### ISSUE THREAD

**Issue by Preclowski** [FriendsOfSymfony/FOSRestBundle#1887]
Hi, I think `twig` dependency should be moved from require-dev to require or made optional some way.

Even if we use custom exception controller, there is registered service `fos_rest.exception.twig_controller` with dependency to `templating.engine.twig`.

```xml
        <service id="fos_rest.exception.twig_controller" class="FOS\RestBundle\Controller\TwigExceptionController" parent="fos_rest.exception.controller">
            <argument type="service" id="templating.engine.twig" />
        </service>
```

It would be nice if twig will not be required to use fosrest 😄

**Comment by stof** [comment]
but is this service *used* when Twig is not available ?

**Comment by Preclowski** [comment]
I dont think so, `fos_rest.exception.controller` should be used by default. But it's declared and symfony throws exception 

> The service "fos_rest.exception.twig_controller" has a dependency on a non-
    existent service "templating.engine.twig".

**Comment by xabbuh** [comment]
Which version of FOSRestBundle do you use?

**Comment by xabbuh** [comment]
And what does your config look like?

**Comment by magnetik** [comment]
Might be related : https://github.com/FriendsOfSymfony/FOSRestBundle/issues/1772

**Comment by LLCDBDBBSysUser** [comment]
I have the same problem. Fix is available in #1945. Would be nice if it would get merged.
